### ISSUE TITLE
Node modules are causing block-scope errors on `requrie` statements...

### ISSUE THREAD

**Issue by S0AndS0** [microsoft/TypeScript#38282]
## TLDR

Convert `require` to `import` if TypeScript is barfing about `node_modules/@types` declaring something that the target project is also dependent upon.

Time is non-refundable so readers that just want an example of one solution should skip to the "Steps to mitigate" section of this Issue.

## Steps to reproduce

**Setup project commands**

```Bash
git init project-name

cd project-name

npm init

npm install --save-dev typescript
npm install --save-dev '@types/node'
```

**`tsconfig.json`**

```JSON
{
  "compilerOptions": {
    "module": "commonjs",
    "esModuleInterop": true,
    "target": "es6",
    "moduleResolution": "node",
    "sourceMap": false,
    "outDir": "."
  },
  "lib": ["es2015"],
  "include": ["ts/**/*"],
  "exclude": ["node_modules"]
}
```

**`package.json`**

```JSON
{
  "name": "project-name",
  "version": "0.0.1",
  "description": "JavaScript project that does stuff and things",
  "main": "index.js",
  "scripts": {
    "ts-build": "tsc --build"
  },
  "repository": {
    "type": "git",
    "url": "git+ssh://git@github.com/account-name/project-name.git"
  },
  "keywords": [
    "TypeScript",
    "JavaScript",
    "NodeJS",
    "block-scope",
    "Errors"
  ],
  "author": "S0AndS0",
  "license": "AGPL-3.0",
  "bugs": {
    "url": "https://github.com/account-name/project-name/issues"
  },
  "homepage": "https://github.com/account-name/project-name#readme",
  "devDependencies": {
    "@types/node": "^13.13.4",
    "typescript": "^3.8.3"
  },
  "dependencies": {}
}
```

**`ts/index.ts` with `const` and `require`**

```TypeScript
#!/usr/bin/env node


'use strict';


const process = require('process');


console.log(`process.env -> ${process.env}`);
```

**Error log from `npm run ts-build`**

```
node_modules/@types/node/globals.d.ts:143:13 - error TS2451: Cannot redeclare block-scoped variable 'process'.

143 declare var process: NodeJS.Process;
                ~~~~~~~

  ts/index.ts:6:7
    6 const process = require('process');
            ~~~~~~~
    'process' was also declared here.


Found 1 error.
```

**`ts/index.ts` with `const` and `require`**

```TypeScript
#!/usr/bin/env node


'use strict';


var process = require('process');


console.log(`process.env -> ${process.env}`);
```

**Error log from `npm run ts-build`**

```
node_modules/@types/node/globals.d.ts:143:13 - error TS2451: Cannot redeclare block-scoped variable 'process'.

143 declare var process: NodeJS.Process;
                ~~~~~~~

  ts/index.ts:6:5
    6 var process = require('process');
            ~~~~~~~
    'process' was also declared here.


Found 1 error.
```

## Expected behavior

I'd expect that `const` declared in one scope (or in this case a totally different file) to not effect another.

I'd expect `exclude` to exclude paths, and/or files, and/or directories; note, I did try things like `"exclude": ["node_modules/**/*"]` and other similar incantations, all to no avail.

## Actual behavior

TypeScript errors out with the above messages, and this isn't isolated to `process` alone, quite a few other Node packages cause TypeScript to barf in a similar fashion.

## Playground Link

The link did **not** generate, however things **did** compiled **but** with the following complaint...

```
Cannot find name 'require'. Do you need to install type definitions for node? Try `npm i @types/node`.(2580)
```

## Related Issues

Likely more that quite a few, but after the time spent writing this Issue instead of code I'll have to leave this section for maybe filling out later, maybe.

## Steps to mitigate

**`ts/index.ts` with `import` and `from`**

```TypeScript
#!/usr/bin/env node


'use strict';


import process from 'process';


console.log(`process.env -> ${process.env}`);
```

Success! File is built to `./index.js` as directed... However what is up with `require` and more to the point why does `exclude` not do what's on the tin?!

While I'm relieved that there is some kind of workaround, I find it a bit irksome that yet another MS product has these sorts of quirks.

I've searched through the FAQ, I've searched about on the World Wide Web, and I've searched within the Issues for this repository. But I've not found a satisfactory answer to why `exclude` configuration is ignored, nor why it exists if it'll be ignored regardless, nor why `require` statements must be converted to `import` for NodeJS projects.

**Comment by RyanCavanaugh** [comment]
Files without `import`/`export` statements are interpreted to be "script" files rather than "module" files, thus their declarations are assumed to be in the global scope. It is an error to redeclare a variable in the global scope.

You can enable the `--isolatedModules` flag to ensure that TS treats every implementation file as a module.

**Comment by S0AndS0** [comment]
Thank you!

I had a suspicion that there was some sort of scoping assumptions being made by the compiler, but couldn't find the correct search incantations or configurations to modify this behavior.

I'm pretty sure your suggestion totally solves this Issue; thanks again.
